datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -764,6 +768,7 @@ pub fn operator_to_name(op: Operator) -> &'static str {
     }
 }
 
+#[allow(deprecated)]
```

Comment:
is this deprecated use required for changes in the substrait dependency?

Reply:
I think so.. `grouping_expressions` field was deprecated in the proto definitions, but should still be used for some time not to break backwards-compatibility. Maybe there's some other way.. I just saw similar deprecated markers and followed suit.

Reply:
Yeah, it's not a rush, and might be better to stick with the old approach for now for producers until consumers have had a chance to catch up.

The new approach puts the `grouping_expressions` in the `AggregateRel` (so in `to_substrait_agg_measure`) and not in the `Grouping`.  Instead, the `Grouping` has a list of indices into the `AggregateRel`'s `grouping_expressions`.

E.g. instead of...

```
AggregateRel = {
  "groupings": [
    { "grouping_expressions": [expr_1, epxr_2] }
  ]
}
```

You would have...

```
AggregateRel = {
  "grouping_expressions": [expr_1, expr_2],
  "groupings": [
    { "expression_references": [0, 1] }
   ]
}
```

This makes it easier to recognize something like a rollup:

```
AggregateRel = {
  "grouping_expressions": [expr_1, expr_2],
  "groupings": [
    { "expression_references": [0, 1] },
    { "expression_references": [0] },
    { "expression_references": [] }
   ]
}
```